### PR TITLE
(feat) O3-3178 : Add a reusable page header component to esm-styleguide

### PR DESCRIPTION
**Pull request by Twiineenock**
# Requirements

- [ ] This PR has a title that briefly describes the work done including the ticket number. Ensure your PR title includes a [conventional commit](https://o3-docs.openmrs.org/docs/frontend-modules/contributing#contributing-guidelines) label (such as `feat`, `fix`, or `chore`, among others). See existing PR titles for inspiration.

## For changes to apps

- [ ] My work conforms to the [**O3 Styleguide**](https://om.rs/styleguide) and [**design documentation**](https://om.rs/o3ui).

## If applicable

- [ ] My work includes tests or is validated by existing tests.
- [ ] I have updated the [esm-framework mock](https://github.com/openmrs/openmrs-esm-core/blob/main/packages/framework/esm-framework/mock.tsx) to reflect any API changes I have made.

## Summary

The `page-header` is re-built multiple times across pages. It should be a re-usable UI / code component as suggested by **Ciarán Duffy** here [https://openmrs.atlassian.net/browse/O3-3178](https://openmrs.atlassian.net/browse/O3-3178)

## Screenshots

page-header
As you can see, all the patient managements have a similar page header

## Related Issue

[https://openmrs.atlassian.net/browse/O3-3178](https://openmrs.atlassian.net/browse/O3-3178)

## Other

In this PR, the date and site info are not implemented into the component as **Ciarán** even suggested we can do away with the date. Though this is still subject to insights from the designs team and reviewers

### CODE REVIEW

File: packages/framework/esm-styleguide/src/page-header/page-header-component.tsx
Comment (Twiineenock) [inline packages/framework/esm-styleguide/src/page-header/page-header-component.tsx:1]: `const combineWords = (sentence: string): string => {
    const lowerCaseWords = (match: string) => match.toLowerCase();
    const capitalizeLetter = (_, group1: string) => group1.toUpperCase();
    return sentence.replace(/\b\w+/g, lowerCaseWords).replace(/\s(\w)/g, capitalizeLetter);
  };`                                                                                                                                                                                                               The `combineWords` function converts a string e.g `Patient lists` to `patientLists` hence changing a string to camel-case to provide the first parameter of the translations `t` function.

Comment (Twiineenock) [inline packages/framework/esm-styleguide/src/page-header/page-header-component.tsx:1]: The `<ExtensionSlot name="page-header-utilities-slot" className={styles.rightJustifiedItems} />` slot provides a place for page headers with **Unique** details e.g the `New list +` button in the `patient lists` and other details a page header.

**Comment by brandones** [comment]
Hi @Twiineenock , please add links to PRs on esm-home (or another relevant repository) showing the usage of this component. Along with screenshots that show how it looks in use.

**Comment by Twiineenock** [comment]
> Hi @Twiineenock , please add links to PRs on esm-home (or another relevant repository) showing the usage of this component. Along with screenshots that show how it looks in use.

Hello @brandones,

I haven't initiated PRs to other repositories yet because they need to import this component from esm-core. I've manually tested the component by integrating it into esm-home as a replacement for the current header. I will have to do that after the merge into the core!.

**Comment by brandones** [comment]
Sounds like you don't have your development environment set up for this kind of work. [Here are the instructions](https://github.com/openmrs/openmrs-esm-core?tab=readme-ov-file#linking-the-framework).

**Comment by brandones** [comment]
Being able to see how the new component actually gets used is an important part of determining whether this is the right API.

**Comment by Twiineenock** [comment]
Alright then, let me work on that.

**Comment by Twiineenock** [comment]
@brandones , I have through the [core readme](https://github.com/openmrs/openmrs-esm-core?tab=readme-ov-file#linking-the-framework), [yarn link](https://classic.yarnpkg.com/en/docs/cli/link) and [yalc](https://www.npmjs.com/package/yalc). But the packages are not linking yet. I raised this in the coffee break and they suggested I link up to you and probably get a reliable solution. And perhaps improve the readme.md. Thanks!

**Pull request closed without merging**